File: common/library-sdk/src/main/java/com/dn/sdk/sdk/interfaces/listener/preload/IAdNativeLoadListener.java
package com.dn.sdk.sdk.interfaces.listener.preload;

import android.view.View;

import java.util.List;

/**
 * 信息流广告
 *
 * @author XuShuai
 * @version v1.0
 * @date 2021/9/26 15:45
 */
public interface IAdNativeLoadListener {

	/**
	 * 加载信息流广告成功
	 *
	 * @param views 信息流View
	 */
	void onLoad(List<View> views);


	/**
	 * 加载广告失败
	 *
	 * @param code  失败code
	 * @param error 错误信息
	 */
	void onLoadFail(int code, String error);


}

File: common/library-sdk/src/main/java/com/dn/sdk/listener/IAdNativeLoadListener.kt
package com.dn.sdk.listener

import com.dn.sdk.bean.natives.INativeAdData

/**
 * 信息流自渲染广告回调监听
 *
 * @author XuShuai
 * @version v1.0
 * @date 2021/12/2 13:55
 */
interface IAdNativeLoadListener {

    /** 无广告填充 */
    fun onAdError(code: Int, errorMsg: String?)

    /** 请求广告成功, 返回数据信息*/
    fun onAdLoad(list: List<INativeAdData>)
}